### PR TITLE
fix broken behaviour when included in an iron-form

### PR DESCRIPTION
**Pull request by notwaldorf**
To make this work in an `iron-form` correctly (https://github.com/PolymerElements/iron-form/issues/12):
- added the `iron-form-element-behaviour` and a `value` property, to be used in serializing. It just mirrors `bind-value`, which I think is vestigial property naming from the days when this element was actually a `<textarea is="iron-textarea">`
- added a `validate()` method, mirrored after `iron-input`. Since `iron-autogrow-textarea` almost never has a validator set, it wasn't really getting validated at all.

The only reason why the `iron-autogrow-textarea` cared about `name` is because it magically worked in an `iron-form`. However, since it's a custom element, we process it's value separately, and don't need to pass it down to the `textarea`. If we do pass it down to the `textarea` then this will be processed twice, since we can't easily differentiate between a textarea that lives inside a `paper-textarea` and a plain boring textarea.

Anyway, the only thing this will break is that `iron-autogrow-textarea` can't be used in a native form, which to be fair is true of all custom elements, so I think it's fine.

Once this lands, we need to remove the `iron-form-element-behaviour` from `paper-textarea`, so that the latter doesn't get added to a form twice: https://github.com/PolymerElements/paper-input/pull/109

### CODE REVIEW

File: iron-autogrow-textarea.html [hunk 1]
```diff
@@ -78,7 +78,6 @@
         autocomplete$="[[autocomplete]]"
         autofocus$="[[autofocus]]"
         inputmode$="[[inputmode]]"
-        name$="[[name]]"
```
Comment (morethanreal) [inline iron-autogrow-textarea.html]: Since `<iron-autogrow-textarea>` doesn't have form behavior, will it still work in a native form in shady dom or `iron-form` without the `name` attribute?

File: iron-autogrow-textarea.html [hunk 2]
```diff
@@ -207,6 +206,30 @@
...
+      if (this.hasValidator()) {
+        valid = Polymer.IronValidatableBehavior.validate.call(this, this.value);
+      } else {
+        valid = this.$.textarea.validity.valid;
+        this.invalid = !valid;
+      }
+      this.fire('iron-input-validate');
+      return valid;
+    },
+
```
Comment (morethanreal) [inline iron-autogrow-textarea.html]: Should this be in a different PR (or change the PR description) since it seems not related to duplicating the name attribute?

File: iron-autogrow-textarea.html (hunk 1 above)
Comment (notwaldorf) [inline iron-autogrow-textarea.html]: It would not work in an `iron-form` in any case, because it's not registering itself. You're right that it wouldn't work in a native form anymore. It still has a `name` property, so I could just add the form behaviour to it, which would make it work in an `iron-form`, and remove it from `paper-textarea` (since that just passes it down), to avoid duplication.

File: iron-autogrow-textarea.html (hunk 2 above)
Comment (notwaldorf) [inline iron-autogrow-textarea.html]: Changed the title and description.

File: iron-autogrow-textarea.html [hunk 3]
```diff
@@ -167,6 +168,15 @@
       },
 
       /**
+       * The value for this input, same as `bindValue`
+       */
+      value: {
+        notify: true,
+        type: String,
+        computed: '_computeValue(bindValue)'
```
Comment (notwaldorf) [inline iron-autogrow-textarea.html]: I think in a future PR we should maybe rename `bind-value` to `value`? It is `bind-value` because this used to be a `<textarea is="iron-textarea">` which is no longer the case.

**Comment by notwaldorf** [comment]
@morethanreal I fixed a whole bunch of things, and updated the description. PTAnotherL? :cat:

File: iron-autogrow-textarea.html (hunk 1 above)
Comment (morethanreal) [inline iron-autogrow-textarea.html]: Ok. The coordination with the form behavior and elements that compose form elements seem complicated. It would be nice to see if there's a cleaner way, but maybe don't need to do that for this PR for now. I trust your best judgement.

File: iron-autogrow-textarea.html (hunk 3 above)
Comment (morethanreal) [inline iron-autogrow-textarea.html]: Perhaps although that would entail a breaking API change. Might be better to keep backwards compatible.

**Comment by morethanreal** [comment]
lgtm
